src/components/LangItem.tsx

```diff
@@ -0,0 +1,94 @@
+import { clx } from "helpers";
+import React, { memo } from "react";
+
+const customNames: { [key: string]: string } = {
+  raku: "perl 6",
+};
+
+const customTopics: { [key: string]: string } = {
+  "c++": "cpp",
+  "c#": "csharp",
+  "f#": "fsharp",
+  "f*": "fstar",
+  "objective-c++": "objective-cpp",
+  "perl 6": "perl6",
+  "ren'py": "renpy",
+  "visual basic .net": "visual-basic-net",
+};
+
+export interface LangItemProps {
+  rank: number;
+  color: string;
+  name: string;
+  diff: number;
+  onClick: (name: string) => void;
+  onMouseOver: (name: string) => void;
+  onMouseOut: () => void;
+  selected?: boolean;
+}
+
+const LangItem: React.FC<LangItemProps> = memo(
```

Comment:
I expect the intent here is to memoize, which is probably a good thing, but I put a `console.log` call in the function here, and it keeps logging on list filtering.

Reply:
Thanks for catching that! I missed the `key` prop when rendering the list.